### PR TITLE
Introduce `/jira subscribe` command to open channel subscriptions modal

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

Since the channel subscriptions modal does not feature everything we think it needs to for all end-users of an organization, we have decided to remove the channel subscriptions button from the channel header for now, and instead have a `/jira subscribe` command to open the modal. This is also an attempt to make it less intrusive for users that have only one plugin button in the channel header.

See the conversation about this decision here: https://community-release.mattermost.com/core/pl/agm7ar97opnf9g83h97pfwhkke

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-17583

I've removed some code that sets up the channel header button, but left the code that the parent component has to hook it up. The former was to make the linter happy, the latter was to make it easier to set up whatever component we need to open the modal in the follow up release, where we improve the channel subscriptions.

### CODE REVIEW

File: webapp/src/hooks/hooks.js [hunk 1]
```diff
@@ -23,6 +23,20 @@ export default class Hooks {
             this.store.dispatch(openCreateModalWithoutPost(description, contextArgs.channel_id));
             return Promise.resolve({});
         }
+
```
Comment (levb) [inline webapp/src/hooks/hooks.js]: nit: and can be a separate PR, but perhaps trim the white space and lowercase the command at the top of the function?

File: server/command.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ const helpText = "###### Mattermost Jira Plugin - Slash Command Help\n" +
 	"* `/jira create <text (optional)>` - Create a new Issue with 'text' inserted into the description field\n" +
 	"* `/jira transition <issue-key> <state>` - Change the state of a Jira issue\n" +
 	"* `/jira view <issue-key>` or `/jira <issue-key>` - View a Jira issue\n" +
+	"* `/jira subscribe` - Create or edit a subscription for this channel\n" +
```
Comment (cpoile) [inline server/command.go]: @aaronrothschild suggest: `Create or edit a Jira webhook subscription for this channel`?

Comment (aaronrothschild) [reply server/command.go]:
```suggestion
	"* `/jira subscribe` - Configure the Jira notifications sent to this channel \n" +
```
Since it's opening a modal, we don't need any commands.

**Comment by DHaussermann** [comment]
Unsure if Dev review label is still needed. I re-added it as I had removed it by mistake.

File: server/command.go (hunk 2 above)
Comment (levb) [reply server/command.go]: Hmmm, shouldn't then the command be `/jira notify`? I know... I know...

**Comment by levb** [comment]
I see all approvals, and no labels removed. @mickmister resolve the conflict and merge?
